cmd/fyne/internal/commands/translate.go

```diff
@@ -0,0 +1,374 @@
+package commands
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/urfave/cli/v2"
+)
+
+// Translate returns the cli command to scan for new translation strings.
+func Translate() *cli.Command {
+	return &cli.Command{
+		Name:  "translate",
+		Usage: "Scans for new translation strings.",
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name:    "verbose",
+				Aliases: []string{"v"},
+				Usage:   "Show files that are being scanned etc.",
+			},
+			&cli.BoolFlag{
+				Name:    "update",
+				Aliases: []string{"u"},
+				Usage:   "Update existing translations (use with care).",
+			},
+			&cli.BoolFlag{
+				Name:    "dry-run",
+				Aliases: []string{"n"},
+				Usage:   "Scan without storing the results.",
+			},
+			&cli.StringFlag{
+				Name:    "sourceDir",
+				Aliases: []string{"d"},
+				Usage:   "Directory to scan recursively for go files.",
+				Value:   ".",
+			},
+			&cli.StringFlag{
+				Name:    "translationsFile",
+				Aliases: []string{"f"},
+				Usage:   "File to read from and write translations to.",
+				Value:   "translations/en.json",
+			},
+		},
+		Action: func(ctx *cli.Context) error {
+			sourceDir := ctx.String("sourceDir")
+			translationsFile := ctx.String("translationsFile")
+			files := ctx.Args().Slice()
+			opts := translateOpts{
+				DryRun:  ctx.Bool("dry-run"),
+				Update:  ctx.Bool("update"),
+				Verbose: ctx.Bool("verbose"),
+			}
+
+			// without any argument find all .go files in the source directory
+			if len(files) == 0 {
+				sources, err := findFilesExt(sourceDir, ".go")
+				if err != nil {
+					return err
+				}
+				files = sources
+			}
+
+			// update the translation file by scanning the given source files
```

Comment:
We prefer to omit inline documentation for the most part - here it is clear from the factored our function name what is happening